Makefile

```diff
@@ -4,43 +4,32 @@
 mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
 mkfile_dir := $(dir $(mkfile_path))
 
-export TARGET_ARCH ?= $(shell uname -m | sed s/x86_64/amd64/ | sed s/aarch64/arm64/)
-export TARGET_TRIPLE ?= $(shell command -v rustc 1>/dev/null && rustc -vV | sed -n 's|host: ||p')
-
-# High-level settings that ultimately get passed down to build-specific targets.
-export APP_FULL_NAME ?= Agent Data Plane
-export APP_SHORT_NAME ?= data-plane
-export APP_IDENTIFIER ?= adp
-export APP_GIT_HASH ?= $(shell git rev-parse --short HEAD 2>/dev/null || echo not-in-git)
-export APP_VERSION ?= $(shell cat bin/agent-data-plane/Cargo.toml | grep -E "^version = \"" | head -n 1 | cut -d '"' -f 2)
+# High-level settings that ultimately get passed down to build-specific targets and other spots.
+export TARGET_ARCH := $(shell uname -m | sed s/x86_64/amd64/ | sed s/aarch64/arm64/)
+export BUILD_TARGET := $(or $(BUILD_TARGET),$(shell rustc -vV | sed -n 's|host: ||p'))
+export APP_GIT_HASH := $(or $(CI_COMMIT_SHA),$(shell git rev-parse --short HEAD 2>/dev/null || echo not-in-git))
+export APP_BUILD_TIME := $(or $(CI_PIPELINE_CREATED_AT),0000-00-00T00:00:00-00:00)
+
+# ADP-specific settings used during builds.
+export ADP_APP_FULL_NAME := Agent Data Plane
+export ADP_APP_SHORT_NAME := data-plane
+export ADP_APP_IDENTIFIER := adp
+export ADP_APP_GIT_HASH := $(APP_GIT_HASH)
+export ADP_APP_VERSION_AUTO := $(shell cat bin/agent-data-plane/Cargo.toml | grep -E "^version = \"" | head -n 1 | cut -d '"' -f 2)
+export ADP_APP_VERSION := $(or $(ADP_APP_VERSION),$(ADP_APP_VERSION_AUTO))
+export ADP_APP_BUILD_TIME := $(APP_BUILD_TIME)
+
+# General build settings used for tooling, etc.
+export GO_BUILD_IMAGE ?= golang:1.23-bullseye
+export GO_APP_IMAGE ?= ubuntu:24.04
 
-# Override autoinstalling of tools. (Eg `cargo install`)
+# Tool configuration.
 export AUTOINSTALL ?= true
+export CARGO_BIN_DIR := $(shell echo "${HOME}/.cargo/bin")
 export CARGO_BINSTALL_STRATEGIES ?= crate-meta-data,compile
 ifeq ($(CI),true)
 	override CARGO_BINSTALL_STRATEGIES = compile
 endif
-# Override the container tool. Tries docker first and then tries podman.
-export CONTAINER_TOOL ?= auto
-ifeq ($(CONTAINER_TOOL),auto)
-	ifeq ($(shell docker version >/dev/null 2>&1 && echo docker), docker)
-		override CONTAINER_TOOL = docker
-	else ifeq ($(shell podman version >/dev/null 2>&1 && echo podman), podman)
-		override CONTAINER_TOOL = podman
-	else
-		override CONTAINER_TOOL = unknown
-	endif
-endif
```

Comment:
Removed this (and switched all usages of it to `docker`) because Docker is ubiquitous enough that it was just leading to clutter in the file.

If people feel strongly about being able to use another compatible equivalent like `podman`, or something else... we can always add it back. 🤷🏻